tahoe_idp/magiclink_helpers.py

```diff
@@ -0,0 +1,36 @@
+from datetime import timedelta
+
+from django.conf import settings as djsettings
+from django.http import HttpRequest
+from django.utils import timezone
+from django.utils.crypto import get_random_string
+from tahoe_idp import magiclink_settings as settings
+from tahoe_idp.magiclink_utils import get_url_path
+from tahoe_idp.models import MagicLink, MagicLinkError
+
+
+def create_magiclink(
+    username: str,
+    request: HttpRequest,
+    redirect_url: str = '',
+) -> MagicLink:
+    limit = timezone.now() - timedelta(seconds=settings.LOGIN_REQUEST_TIME_LIMIT)  # NOQA: E501
+    over_limit = MagicLink.objects.filter(username=username, created_on__gte=limit)
+    if over_limit:
+        raise MagicLinkError('Too many magic login requests')
+
+    # Only the last magic link is usable per user
+    MagicLink.objects.filter(username=username, used=False).update(used=True)
+
+    if not redirect_url:
+        redirect_url = get_url_path(djsettings.LOGIN_REDIRECT_URL)
```

Comment:
Let's not set default and let Studio take care of defaulting to /home.

This helps to make LMS unconcerned about Studio details.

Studio and LMS don't share the same `LOGIN_REDIRECT_URL` therefore this will probably break in production.

```suggestion
```